app/src/main/java/org/dhis2/usescases/notes/NotesFragment.kt

```diff
@@ -94,18 +96,29 @@ class NotesFragment : FragmentGlobalAbstract(), NotesView {
         savedInstanceState: Bundle?
     ): View? {
         binding = DataBindingUtil.inflate(inflater, R.layout.fragment_notes, container, false)
-        noteAdapter = NotesAdapter()
+        noteAdapter = NotesAdapter(this)
         binding.notesRecycler.adapter = noteAdapter
         binding.addNoteButton.setOnClickListener {
-            val args = Bundle()
-            args.putString(Constants.PROGRAM_UID, programUid)
-            args.putString(Constants.UID, uid)
-            args.putSerializable(Constants.NOTE_TYPE, noteType)
-            // TODO: startActivity with args
+            val intent = Intent(activity, NoteDetailActivity::class.java).apply {
+                putExtra(Constants.PROGRAM_UID, programUid)
+                putExtra(Constants.UID, uid)
+                putExtra(Constants.NOTE_TYPE, noteType)
+            }
+            startActivity(intent)
```

Comment:
You could also do :+1: 
Intent(activity, NoteDetailActivity::class.java).apply {
                putExtra(Constants.PROGRAM_UID, programUid)
                putExtra(Constants.UID, uid)
                putExtra(Constants.NOTE_TYPE, noteType)
            }. also {startActivity(it))